lib/VeresOneDidDoc.js

```diff
@@ -384,99 +385,102 @@ class VeresOneDidDoc {
           }
         }
       ],
+      patch,
+      sequence,
       target: this.id,
-      sequence: this.meta.sequence,
-      patch
     };
   }
 
   /**
    * Composes and returns a service id for a service name.
    *
-   * @param serviceName {string}
+   * @param {string} serviceName
    *
    * @returns {string} Service id
    */
-  serviceIdFor(serviceName) {
-    if(!serviceName) {
-      throw new Error('Invalid service name.');
+  serviceIdFor(fragment) {
+    if(!fragment) {
+      throw new Error('Invalid service fragment.');
     }
-
-    return this.id + ';service=' + encodeURIComponent(serviceName);
+    return `${this.id}#${fragment}`;
   }
 
   /**
    * Finds a service endpoint in this did doc, given an id or a name.
    *
-   * @param [id] {string}
-   * @param [name] {string}
+   * @param {string} [fragment]
+   * @param {string} [id]
    *
    * @returns {object}
    */
-  findService({id, name}) {
-    const serviceId = id || this.serviceIdFor(name);
+  findService({fragment, id}) {
+    const serviceId = id || this.serviceIdFor(fragment);
 
     return jsonld
       .getValues(this.doc, 'service')
       .find(service => service.id === serviceId);
   }
 
   /**
-   * Tests whether this did doc has a service endpoint (by name or id).
-   * One of `id` or `name` is required.
+   * Tests whether this did doc has a service endpoint (by fragment or id).
+   * One of `id` or `fragment` is required.
    *
-   * @param [id] {string}
-   * @param [name] {string}
+   * @param {string} [id]
+   * @param {string} [name]
    *
    * @returns {boolean}
    */
-  hasService({id, name}) {
-    return !!this.findService({id, name});
+  hasService({id, fragment}) {
+    return !!this.findService({id, fragment});
   }
 
   /**
    * Adds a service endpoint to this did doc.
    * One of `id` or `name` is required.
    *
-   * @param [id] {string}
-   * @param [name] {string}
-   * @param type {string} (e.g. 'AgentService')
-   * @param serviceEndpoint {string} uri (e.g. 'https://agent.example.com')
-   * @param [options] {object} Any additional properties of endpoint
+   * @param {string} [fragment]
+   * @param {string} [id]
+   * @param {string} type URI (e.g. 'urn:AgentService')
+   * @param {string} endpoint  URI (e.g. 'https://agent.example.com')
    */
-  addService({id, name, type, serviceEndpoint, ...options}) {
-    const serviceId = id || this.serviceIdFor(name);
+  addService({fragment, endpoint, id, type}) {
+    if(!!id === !!fragment) {
+      throw new Error('Exactly one of `fragment` or `id` is required.');
+    }
+    if(id && !id.includes(':')) {
```

Comment:
At some point should probably make these a bit more solid.  Check the part before `:` is just valid scheme chars.  Probably not a big deal at the moment.